Sources/TensorUtils/LogitsWarper/LogitsWrapper.swift

```diff
@@ -0,0 +1,6 @@
+import Foundation
+
+/// Protocol for all logit warpers that can be applied during generation
+public protocol LogitsWarper {
+    func callAsFunction(_ arr: [Float]) -> (indexes: [Int], logits: [Float])
```

Comment:
This is great, but out of personal preference I'd also include a method with a semantically descriptive name of what it does, and then use `callAsFunction` for convenience.

So in this case we'd have a method called `warp`, and then we'd have a default implementation for `callAsFunction` that would simply invoke `warp`. See for example how it's done here: https://github.com/huggingface/swift-transformers/blob/fa2522162afa259fe87246ffa41dec9e87e1f195/Sources/Tokenizers/Decoder.swift#L11-L22